src/libs/LangChain.Core/Memory/FileChatMessageHistory.cs

```diff
@@ -4,36 +4,28 @@
 namespace LangChain.Memory;
 
 /// <summary>
-/// Stores history in a local file.
+/// Chat message history that stores history in a local file.
 /// </summary>
 public class FileChatMessageHistory : BaseChatMessageHistory
 {
     private string MessagesFilePath { get; }
 
-    private List<Message> _messages;
+    private List<Message> _messages = new List<Message>();
 
     /// <inheritdoc/>
-    public override IReadOnlyList<Message> Messages
-    {
-        get
-        {
-            if (_messages is null)
-            {
-                LoadMessages().Wait();
-            }
+    public override IReadOnlyList<Message> Messages => _messages;
 
-            return _messages;
-        }
-    }
-    
     /// <summary>
-    /// 
+    /// Initializes new history instance with provided file path
     /// </summary>
-    /// <param name="messagesFilePath">Path to local history file</param>
+    /// <param name="messagesFilePath">path of the local file to store the messages</param>
     /// <exception cref="ArgumentNullException"></exception>
     public FileChatMessageHistory(string messagesFilePath)
     {
         MessagesFilePath = messagesFilePath ?? throw new ArgumentNullException(nameof(messagesFilePath));
+
+        // Blocking call in the constructor creates a simpler implementation
```

Comment:
A very dangerous thing. While this may work in tests, it will likely cause a lot of problems in desktop applications.

Reply:
Good call.  Maybe make the constructor private and using a static factory method?

```
    private FileChatMessageHistory(string messagesFilePath)
    {
        MessagesFilePath = messagesFilePath ?? throw new ArgumentNullException(nameof(messagesFilePath));
    }

    public static async Task<FileChatMessageHistory> Create(string messagesFilePath)
    {
        FileChatMessageHistory chatHistory = new FileChatMessageHistory(messagesFilePath);
        await chatHistory.LoadMessages().ConfigureAwait(false);

        return chatHistory;
    }
```

Reply:
Yes, looks good. Please use CreateAsync and CancellationToken cancellationToken = default as last parameter.
`messagesFilePath` can be just `path`? I think there will be no misunderstanding here due to the context